README.md

```diff
@@ -1,526 +1,917 @@
-- No new namespace or type of second-class value is added--decorators are functions.
+1. They can **replace** the value that is being decorated with a _matching_ value that has the same semantics. (e.g. a decorator can replace a method with another method, a field with another field, a class with another class, and so on).
+2. They can associate **metadata** with the value that is being decorated. This metadata can then be read externally and used for metaprogramming and introspection.
+3. They can provide **access** to the value that is being decorated, via metadata. For public values, they can do this via the name of the value. For private values, they receive accessor functions which they can then choose to share.
 
-# Examples
+Essentially, decorators can be used to metaprogram and add functionality to a value, without fundamentally changing its external behavior.
 
-A few examples of how to implement and use decorators in this proposal:
+This proposal differs from previous iterations where decorators could replace the decorated value with a completely different type of value. The requirement for decorators to only replace a value with one that has the same semantics as the original value fulfills two major design goals:
 
-## `@logged`
+- **It should be easy both to use decorators and to write your own decorators.** Previous iterations such as the _static decorators_ proposal were complicated for authors and implementers in particular. In this proposal, decorators are plain functions, and are accessible and easy to write.
+- **Decorators should affect the thing they're decorating, and avoid confusing/non-local effects.** Previously decorators could change the decorated value in unpredictable ways, and also add completely new values which were unrelated. This was problematic both for _runtimes_, since it meant decorated values could not be analyzed statically, and for _developers_, since decorated values could turn into completely different types of values without any indicator to the user.
 
-The `@logged` decorator logs a console message when a method starts and finishes. Many other popular decorators will also want to wrap a function, e.g., `@deprecated`, `@debounce`, `@memoize`, etc.
+In this proposal, decorators can be applied to the following existing types of values:
 
-Usage:
+- Classes
+- Class fields (public, private, and static)
+- Class methods (public, private, and static)
+- Class accessors (public, private, and static)
 
-```mjs
-import { logged } from "./logged.mjs";
+In addition, this proposal introduces two new types of class elements that can be decorated:
 
-class C {
-  @logged
-  m(arg) {
-    this.#x = arg;
+- Class _properties_, defined by applying the `prop` keyword to a class field. These have a getter and setter, unlike fields, which default to getting and setting the value on a private storage slot (equivalent to a private class field):
```

Comment:
Slight ambiguity from the phrasing here: it currently reads as though *fields* default t o getting and setting on a private storage slot.

```suggestion
- Class _properties_, defined by applying the `prop` keyword to a class field. Unlike fields, these have a getter and setter, which default to getting and setting the value on a private storage slot (equivalent to a private class field):
```